splice_machine/src/main/java/com/splicemachine/derby/impl/stats/StoreCostControllerImpl.java

```diff
@@ -173,8 +178,26 @@ public StoreCostControllerImpl(TableDescriptor td, ConglomerateDescriptor conglo
                     throw StandardException.plainWrapException(e);
                 }
             }
+            useRealTableStatistics = false;
         } else {
             tableStatistics = new TableStatisticsImpl(tableId, partitionStats,fallbackNullFraction,extraQualifierMultiplier);
+            useRealTableStatistics = true;
+            assert !partitionStatistics.isEmpty();
+
+            List<ColumnStatisticsDescriptor> columnStats = partitionStatistics.get(0).getColumnStatsDescriptors();
```

Comment:
Ideally, missing column stats should not be all the columns in the table that do not have stats. It should only be the columns that are used by optimizer to estimate selectivity or cost. For example, 
```
create table t1 (a1 int, b1 int, c1 int);
create table t2(a2 int, b2 int, c2 int);

call syscs_util.disable_column_statistics('SPLICE', 'T1', 'C1');
analyze table t1;
analyze table t2;
explain
select * from t1, t2 where a1=a2 and b1=1;
```
For this query, we don't care whether there is stats on c1 and b2, c2, we only care if there is stats on b1, a1, and a2 as they are the ones used by optimizer. So we should only report missing stats on these columns, but not on c1,  b2, and c2.

Reply:
So ideally, the missing column or stats info is specific to a query, and the info should be stored in the FromBaseTable instances instead of StoreCostControllerImpl which is shared among all the instances of the same table with in the same session.